avocado/core/loader.py

```diff
@@ -44,6 +44,41 @@
 ALL = True
 
 
+def filter_test_tags(test_suite, filter_by_tags):
+    '''
+    The filtering mechanism is limited to instrumented tests with tags set
+    '''
+    filtered = []
+    for klass, info in test_suite:
+        if not isinstance(klass, str):  # not instrumented: no filtering
+            filtered.append((klass, info))
+            continue
+
+        test_tags = info['tags']
+        if not test_tags:       # not tagged: no filtering
```

Comment:
Yep, but if we do: https://github.com/avocado-framework/avocado/pull/1621/commits/5ce9ac2b81b7579443f3f339f9395787f77c041d#r88693977 we can distinguish, am I right?